samples/shields/npm6001_ek/doc/index.rst

```diff
@@ -0,0 +1,104 @@
+.. _npm6001-ek-sample:
+
+nPM6001 EK sample
+#################
+
+Overview
+********
+
+This sample is provided as an example to test the :ref:`npm6001_ek`. The sample
+provides a shell interface that allows to test multiple functionalities offered
+by the nPM6001 PMIC, including:
+
+- GPIO
+- Watchdog
```

Comment:
I actually had the same feeling, and indeed, I think it is the correct way to approach this, since we're in the end implementing generic drivers. Let me take a look and see how we can support vendor-specific features as well.

Reply:
I've actually checked and usability of shells has really degraded after the removal of labels. In case of MFD, sub-nodes receive names like `gpio-controller` (names suggested by DT spec), so when listing devices you can easily get 2 that have the same name. We really need to fix that. One way could be to use device path, and shells ask for it interactively before doing any operation.

I'm inclined to leave this sample for now, and try to address gpio/wdt/other shells in the near future. Thoughts?

Reply:
I am okay with that.